### ISSUE TITLE
Failed to parse json generated by vswhere

### ISSUE THREAD

**Issue by kimura-synthesis** [vector-of-bool/vscode-cmake-tools#220]
I got following error with Windows  10(Japanese) and MSVC 2017 Community.

```
Unexpected token � in JSON at position 368: SyntaxError: Unexpected token � in JSON at position 368
    at JSON.parse (<anonymous>)
    at Object.<anonymous> (C:\Users\kkimura\.vscode\extensions\vector-of-bool.cmake-tools-0.10.2\out\src\environment.js:271:39)
    at Generator.next (<anonymous>)
    at fulfilled (C:\Users\kkimura\.vscode\extensions\vector-of-bool.cmake-tools-0.10.2\out\src\environment.js:4:58)
```
Here is output of `vswhere.exe -all -format json -product *  -legacy`.
```json
[
  {
    "instanceId": "01eb69bb",
    "installDate": "2017-07-31T06:40:09Z",
    "installationName": "VisualStudio/15.2.0+26430.16",
    "installationPath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community",
    "installationVersion": "15.0.26430.16",
    "displayName": "Visual Studio Community 2017",
    "description": "�w���A�I�[�v�� �\�[�X�A����ьX�̊J���҂̂��߂̖����Ŋ��S�ȋ@�\������� IDE",
    "enginePath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\resources\\app\\ServiceHub\\Services\\Microsoft.VisualStudio.Setup.Service",
    "channelId": "VisualStudio.15.Release",
    "channelPath": "C:\\Users\\kkimura\\AppData\\Local\\Microsoft\\VisualStudio\\Packages\\_Channels\\4CB340F5\\catalog.json",
    "channelUri": "https://aka.ms/vs/15/release/channel",
    "releaseNotes": "https://go.microsoft.com/fwlink/?LinkId=660284#15.1.26430.16",
    "thirdPartyNotices": "https://go.microsoft.com/fwlink/?LinkId=660300"
  }
]
```

Output encoding is CP932. I think this is vswhere's bug.

**Comment by vector-of-bool** [comment]
Just opened a ticket for vswhere. We'll see what happens.

**Comment by kimura-synthesis** [comment]
I've succeeded generate json file encoded with UTF-8.

Following typescript code is for your reference.
```typescript
import * as proc from 'child_process'

function runVSWhere(){
  let vswhere:string = '/path/to/vswhere.exe';
  proc.exec(`powershell.exe -Command "& '${vswhere}' -format json | Set-Content -Encoding UTF8 tmp.json"`);
}
```
`tmp.json` is belows.
```json
[
  {
    "instanceId": "01eb69bb",
    "installDate": "2017-07-31T06:40:09Z",
    "installationName": "VisualStudio/15.2.0+26430.16",
    "installationPath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community",
    "installationVersion": "15.0.26430.16",
    "displayName": "Visual Studio Community 2017",
    "description": "学生、オープン ソース、および個々の開発者のための無料で完全な機能を備えた IDE",
    "enginePath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\resources\\app\\ServiceHub\\Services\\Microsoft.VisualStudio.Setup.Service",
    "channelId": "VisualStudio.15.Release",
    "channelPath": "C:\\Users\\kkimura\\AppData\\Local\\Microsoft\\VisualStudio\\Packages\\_Channels\\4CB340F5\\catalog.json",
    "channelUri": "https://aka.ms/vs/15/release/channel",
    "releaseNotes": "https://go.microsoft.com/fwlink/?LinkId=660284#15.1.26430.16",
    "thirdPartyNotices": "https://go.microsoft.com/fwlink/?LinkId=660300"
  }
]
```

**Comment by heaths** [comment]
Just got another report of this at Microsoft/vswhere#115, but you don't need to exec to powershell (rather heavy). `child_process.execFile` accepts an [options with an encoding](https://nodejs.org/api/child_process.html#child_process_child_process_exec_command_options_callback). Additionally, by using `child_process.exec` you're actually running vswhere through cmd.exe which is exacting its own code page handling. Instead, `child_porocess.execFile` would be direct execution. Using `execFile` I was able to make this work.

You could also use `child_process.spawn` and get a direct stream. If your JSON parser supports a `buffer` that would eliminate encoding (rather, keep the original UTF8 that vswhere gets from the query API).

Whichever way, `exec` is probably not the best choice since it goes through cmd.exe which has its own code page hassles.

**Comment by CHChang810716** [comment]
seems same problem
```
CMakeTools extension was unable to initialize: SyntaxError: Unexpected token � in JSON at position 562
```
vswhere output:
```
[
  {
    "instanceId": "a95aa3c0",
    "installDate": "2017-12-14T10:57:48Z",
    "installationName": "VisualStudio/15.5.2+27130.2010",
    "installationPath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community",
    "installationVersion": "15.5.27130.2010",
    "productId": "Microsoft.VisualStudio.Product.Community",
    "productPath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\Common7\\IDE\\devenv.exe",
    "isPrerelease": false,
    "displayName": "Visual Studio Community 2017",
    "description": "功能齊備的 IDE，可供學生、開放原始碼開發人員與個人開發人員免費使用",
    "channelId": "VisualStudio.15.Release",
    "channelPath": "C:\\Users\\Colleta55\\AppData\\Local\\Microsoft\\VisualStudio\\Packages\\_Channels\\4CB340F5\\catalog.json",
    "channelUri": "https://aka.ms/vs/15/release/channel",
    "enginePath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\resources\\app\\ServiceHub\\Services\\Microsoft.VisualStudio.Setup.Service",
    "releaseNotes": "https://go.microsoft.com/fwlink/?LinkId=660692#15.5.2",
    "thirdPartyNotices": "https://go.microsoft.com/fwlink/?LinkId=660708",
    "catalog": {
      "buildBranch": "d15svc",
      "buildVersion": "15.5.27130.2010",
      "id": "VisualStudio/15.5.2+27130.2010",
      "localBuild": "build-lab",
      "manifestName": "VisualStudio",
      "manifestType": "installer",
      "productDisplayVersion": "15.5.2",
      "productLine": "Dev15",
      "productLineVersion": "2017",
      "productMilestone": "RTW",
      "productMilestoneIsPreRelease": "False",
      "productName": "Visual Studio",
      "productPatchVersion": "2",
      "productPreReleaseMilestoneSuffix": "0.0",
      "productRelease": "RTW",
      "productSemanticVersion": "15.5.2+27130.2010"
    },
    "properties": {
      "campaignId": "1769436087.1498238982",
      "channelManifestId": "VisualStudio.15.Release/15.5.2+27130.2010",
      "nickname": "",
      "setupEngineFilePath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\vs_installershell.exe"
    }
  }
]
```

**Comment by vector-of-bool** [comment]
I've pushed out a possible fix into the beta5 for 0.11.0. If you find that the fix does not work, please re-open this issue. We've had trouble reproducing it so aren't 100% certain that our fix works.
